includes/course-video/class-sensei-course-video-settings.php

```diff
@@ -90,6 +93,27 @@ public function add_feature_flag_inline_script() {
 		}
 	}
 
+	/**
+	 * Enqueues settings scripts on the frontend.
+	 *
+	 * @return void
+	 */
+	public function enqueue_settings_scripts() {
+		if ( is_admin() || get_post_type() !== 'lesson' ) {
+			return;
+		}
+
+		$video_settings      = [
+			'courseVideoAutoComplete' => (bool) $this->is_autocomplete_enabled(),
+			'courseVideoAutoPause'    => (bool) $this->is_autopause_enabled(),
+			'courseVideoRequired'     => (bool) $this->is_required(),
+		];
+		$video_settings_json = wp_json_encode( $video_settings );
+		$script              = "window.sensei = window.sensei || {}; window.sensei.courseVideoSettings = $video_settings_json;";
+
+		wp_add_inline_script( 'sensei-course-video-blocks-extension', $script, 'before' );
```

Comment:
Doesn't appear on the page :( What did I wrong?

Reply:
I found the issue. This code relies on a logic of the Frontend class that is instantiated later.
I changed the order of instantiating and now it works.
However, I think I need to move this asset registration to the Settings class.

Reply:
We're still enqueueing the file on the filter that calls the wrap method.
Probably if we do it under the `wp_enqueue_scripts` action too, it would work? So if we need to change the execution order, we can just change the priority of the hook. ;)

Reply:
https://developer.wordpress.org/reference/functions/wp_add_inline_script/#description

> Code will only be added if the script is already in the queue.

So I think we need to run it after the enqueue.